### DISCUSSION TITLE
Trying to run a scheduled dag with root privileges/as sudo

### DISCUSSION THREAD

**Discussion by trumandaniels** [apache/airflow#23707] [Q&A]
Hi, I'm very very new to Airflow, and trying to figure out how to run a scheduled bash command with root/sudo privileges.

My code looks like this:

```
with DAG(
    dag_id='change_vpn_server',
    schedule_interval='*/10 * * * *', #every 10 minutes
    start_date=datetime(year=2022, month=5, day=12),
    catchup=False,
) as dag:
    task_get_datetime = BashOperator(
        task_id='change_vpn_server_task',
        bash_command='vpn --traffic --country-code US --city "New York" --connect',
        run_as_user='root',
    )
```

PermissionError: [Errno 1] Operation not permitted: '/tmp/tmp_b_5dvd3'

Bonus question: is the airflow webserver publicly facing? i.e. could someone access my dags remotely if they had my login credentials? Or would I need to do additional setup to allow remote login?

**Comment by potiuk** [comment]
You need to setup your user to be able to sudo I believe https://airflow.apache.org/docs/apache-airflow/stable/security/workload.html 

and public websever - - it depends how you deploy airlfow. Airflow is just set of components to run - you need to deploy it in the way you find best https://airflow.apache.org/docs/apache-airflow/stable/installation/index.html and each choice has its own way of exposing/deploying the webserver - airflow webserver listens on specific port - but how you run and expose any of that is entirely up to you - and only you can answer the question if your ports are exposed and how.
